Lab1/app/src/main/java/nm/lab1/MainActivity.java

```diff
@@ -0,0 +1,144 @@
+package nm.lab1;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+
+public class MainActivity extends ActionBarActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        //define parts of interface
+        Button button = (Button) findViewById(R.id.button);
+        final EditText textbox = (EditText) findViewById(R.id.editText);
+        ListView list = (ListView) findViewById(R.id.list);
+
+        final ArrayList<String> arrayList = new ArrayList<String>();
+        final ArrayAdapter<String> listAdapter = new ArrayAdapter<String>(this,
+                android.R.layout.simple_list_item_1, arrayList);
+
+        list.setAdapter(listAdapter);
+
+        //click button listener
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String input = textbox.getText().toString();
+                if(null != input && input.length() > 0){
+                    arrayList.add(input);
+                    listAdapter.notifyDataSetChanged();
+                }
+            }
+        });
+        setupListViewListener(list, arrayList, listAdapter);
+    }
+
+    private void setupListViewListener(ListView list, final ArrayList<String> arrayList, final ArrayAdapter<String> listAdapter){
```

Comment:
Breaking this out into a separate method definitely helps with the readability of the code.